### PR TITLE
applications: machine_learning: Rename ml_state to ml_app_mode

### PR DESCRIPTION
**Pull request by emob-nordic**
Rename module name: ml_state to ml_app_mode.

NCSDK-12535.

### CODE REVIEW

**Comment by rakons** [comment]
Why in this PR that states it only changes ml_state to ml_app_mode there is a commit that changes the way how settings of the sensor of interest is provided?

**Review by rakons — APPROVED** [review]
The only think I dislike in this PR is that it includes also configuration changes not related to ml_state renaming.

**Comment by emob-nordic** [comment]
> The only think I dislike in this PR is that it includes also configuration changes not related to ml_state renaming.

I can drop off this commit and create separate PR for that.

File: applications/machine_learning/README.rst [hunk 1]
```diff
@@ -540,8 +540,8 @@ The nRF Machine Learning application also uses the following dedicated applicati
   The module runs the machine learning model and provides results only if there is an active subsriber.
   An application module can inform that it is actively listening for results using :c:struct:`ml_result_signin_event`.
 
-``ml_state``
-  The module controls switching between running the machine learning model and forwarding the data.
+``ml_app_mode``
+  The module controls Application mode. It switches between running the machine learning model and forwarding the data.
```
Comment (MarekPieta) [inline applications/machine_learning/README.rst:544]: `application mode`? Why using capital letter here?

File: applications/machine_learning/src/events/ml_app_mode_event.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+static int log_ml_app_mode_event(const struct event_header *eh, char *buf, size_t buf_len)
+{
+	const struct ml_app_mode_event *event = cast_ml_app_mode_event(eh);
+
+	BUILD_ASSERT(ARRAY_SIZE(ml_app_mode_name) == ML_APP_MODE_COUNT);
+	__ASSERT_NO_MSG(event->mode < ML_APP_MODE_COUNT);
+	__ASSERT_NO_MSG(ml_app_mode_name[event->mode] != NULL);
+
+	return snprintf(buf, buf_len, "state: %s", ml_app_mode_name[event->mode]);
```
Comment (MarekPieta) [inline applications/machine_learning/src/events/ml_app_mode_event.c:24]: Keep in mind to align this one with @zycz change related to new way of EM events logging.

**Review by MarekPieta — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -69,6 +69,13 @@ nRF9160: Asset Tracker v2
 * Added support for A-GPS filtered ephemerides.
 * Added new documentation for :ref:`asset_tracker_v2_util_module` and :ref:`api_modules_common`.
 
+nRF Machine Learning (Edge Impulse)
+-----------------------------------
+
+* Updated:
+
+  * Renamed ml_state module to ml_app_mode module.
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:77]:
```suggestion
  * Renamed ``ml_state`` to ``ml_app_mode`` module.
```

Comment (emob-nordic) [reply doc/nrf/releases/release-notes-changelog.rst]: done

**Review by divipillai — APPROVED** [review]

**Pull request merged**